woocommerce-gutenberg-products-block.php

```diff
@@ -14,6 +14,10 @@
  * @internal This file is only used when running the REST API as a feature plugin.
  */
 
+use Automattic\WooCommerce\Blocks\Domain\Bootstrap;
```

Comment:
May be risky doing this before the PHP version check?

Reply:
Nope. The only time this will cause a fatal is if run in a PHP 5.2 environment.  Chances of that are pretty slim considering it requires a perfect storm of also installing this on a site without WooCommerce core installed (if the user downloads the plugin separately) _and_ on a version of WordPress not supporting Gutenberg. [WooCommerce Admin also does this](https://github.com/woocommerce/woocommerce-admin/blob/9a64926d8af836493330236c2a1ce0e18ca064ce/woocommerce-admin.php#L22) and I figured if it was a problem, that would have changed by now (although from blame I see it's a recent change).

Reply:
Whilst it's true Woo core won't load this package < 5.3, as a feature plugin it could be installed and activated regardless. 

Thinking about it a different way, why bother with PHP version compare a few lines down if it's going to break before that code runs anyway? :) Ideally the check should run first to prevent a fatal.

Reply:
> as a feature plugin it could be installed and activated regardless.

Very true, but how likely is it that people will install a plugin that has "blocks" in its name if they aren't running Gutenberg in their WordPress environment?

> Thinking about it a different way, why bother with PHP version compare a few lines down if it's going to break before that code runs anyway?

Because the version compare is for PHP5.6+ which covers things that are only available in PHP5.6+.  Namespaces and imports have been available since PHP5.3 so imo it's an edgecase.  With that said, I'll change things to use fully qualified class names instead if you want me to.  If we do so, we should probably be consistent and do the same in WC Admin as well.

Reply:
I prefer to err on the side of caution in the main file, so using fully qualified class names here would be my preference. Once we're sure PHP 5.6 is being used, throughout all other files, we're good to use whatever we need.